spinta/cli/push.py

```diff
@@ -245,69 +370,71 @@ def _attach_keymaps(context: Context, store: Store) -> None:
         context.attach(f'keymap.{keymap.name}', lambda: keymap)
 
 
-def _prepare_rows_for_push(rows: Iterable[ModelRow]) -> Iterator[_PushRow]:
+def _rows_to_push_rows(rows: Iterable[ModelRow]) -> Iterator[_PushRow]:
     for model, row in rows:
-        _id = row['_id']
-        _type = row['_type']
-        where = {
-            'name': 'eq',
-            'args': [
-                {'name': 'bind', 'args': ['_id']},
-                _id,
-            ]
-        }
-        payload = {
-            '_op': 'upsert',
-            '_type': _type,
-            '_id': _id,
-            '_where': spyna.unparse(where),
-            **{k: v for k, v in row.items() if not k.startswith('_')}
-        }
-        yield _PushRow(model, payload)
+        yield _PushRow(model, row)
+
+
+def _prepare_rows_for_push(rows: Iterable[_PushRow]) -> Iterator[_PushRow]:
+    for row in rows:
+        row.data['_op'] = row.op
+        if row.op == 'patch':
+            where = {
+                'name': 'eq',
+                'args': [
+                    {'name': 'bind', 'args': ['_id']},
+                    row.data['_id'],
+                ]
+            }
+            row.data['_where'] = spyna.unparse(where)
+        elif row.op == 'insert':
+            # if _revision is passed insert action gives ManagedProperty error
+            if '_revision' in row.data:
```

Comment:
_revision comes from getall command, I'm not really sure how this could be fixed, because as I understand _revision in only not needed for push, but for other places that use getall command, it is needed.